pkg/provider/aws.go

```diff
@@ -60,34 +66,73 @@ var (
 	}
 )
 
-var awsSurvey = []*survey.Question{
-	{
-		Name:     "cluster",
-		Prompt:   &survey.Input{Message: "Enter the name of your cluster:"},
-		Validate: validCluster,
-	},
+var initAwsSurvey = []*survey.Question{
 	{
 		Name:     "region",
 		Prompt:   &survey.Select{Message: "What region will you deploy to?", Default: "us-east-2", Options: awsRegions},
 		Validate: survey.Required,
 	},
 }
 
+var defaultAwsSurvey = []*survey.Question{
+	{
+		Name:     "cluster",
+		Prompt:   &survey.Input{Message: "Enter the name of your cluster:"},
+		Validate: validClusterName(AWS),
+	},
+}
+
 func mkAWS(conf config.Config) (provider *AWSProvider, err error) {
 	provider = &AWSProvider{}
-	if err = survey.Ask(awsSurvey, provider); err != nil {
+	if err = survey.Ask(initAwsSurvey, provider); err != nil {
+		return
+	}
+
+	var createCluster bool
+	if err = survey.AskOne(&survey.Confirm{Message: "Do you want to create a new cluster?", Default: true}, &createCluster); err != nil {
 		return
 	}
 
 	ctx := context.Background()
 
 	provider.goContext = &ctx
 
-	client, err := getClient(provider.Reg, *provider.goContext)
+	s3Client, err := getClient(provider.Reg, *provider.goContext)
 	if err != nil {
 		return
 	}
 
+	if !createCluster {
+
+		provider.ByokConf = &manifest.ByokConfig{Enabled: !createCluster, Type: EKS}
+
+		eksClient, err2 := getEksClient(provider.Reg, *provider.goContext)
+		if err != nil {
+			return nil, err2
+		}
+
+		input := &eks.ListClustersInput{}
+		clusters, err3 := eksClient.ListClusters(ctx, input)
```

Comment:
we should factor this into a separate method probably to keep this one less sprawling